jetty-core/jetty-fcgi/jetty-fcgi-client/src/main/java/org/eclipse/jetty/fcgi/generator/Flusher.java

```diff
@@ -126,7 +126,7 @@ public Callback join()
             {
                 Throwable cause = getCause();
                 ExceptionUtil.addSuppressedIfNotAssociated(cause, x);
-                ExceptionUtil.ifExceptionThrowUnchecked(cause);
+                ExceptionUtil.throwUnchecked(cause);
```

Comment:
I'm not following this logic.

The exception is caught, with variable `x`.
You `getCause()` from somewhere else, not the `x`.
You add `x` to the mystery `cause` as a suppressed in `cause`.
The local variable `cause` isn't updated.
Then you `throwUnchecked(cause)` ?

First, what is `getCause()` and why not use `x.getCause()`?
Next, if `getCause()` returns null, then the `addSuppressedIfNotAssociated(cause, x)` wont update the `cause` variable.  The same variable you use for `throwUnchecked(cause)`.

Reply:
This PR is not about finding/fixing issues with the flusher.    The new code is the same as the old code: if the wait gets interrupted, then either there is a cause in which case it is thrown if it is uncheck or if it is not, then it is wrapped as un unchecked and thrown.  If the cause is null, a runtime exception is thrown.

But it is a good point that the interrupted exception is lost if the cause is null.  I'll try to fix that at least.